packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -256,6 +269,17 @@ class Auth {
         return this.pendingToken
     }
 
+    /**
+     * Clears the queue. The next queueRequest call will start immediately.
+     * This is used when you call queueRequest recursively to avoid running
+     * into a deadlock.
```

Comment:
```js
        return this.queueRequest(async () => {
            await this.shopperCustomersClient.registerCustomer({
                headers: {
                    authorization: `Bearer ${this.get('access_token')}`
                },
                body
            })
            this.clearQueue()
            return this.loginRegisteredUserB2C({username: email, password}) // <- this is also a queueRequest
        }, isGuest)
```

If there is a nested queueRequest call, the outer queueRequest will never resolve and then the nested queueRequest will never be trigger, thus a deadlock. Open to ideas to systematically fix this implementation, but don't want to spend too much time on this issue either, because I don't imagine this class to grow too much more than what it is today.